src/memray/commands/attach.py

```diff
@@ -298,6 +299,10 @@ def run(self, args: argparse.Namespace, parser: argparse.ArgumentParser) -> None
         verbose = args.verbose
 
         if args.method == "auto":
+            if platform.system() == "Linux" and debugger_available(
+                "gdb", verbose=verbose
+            ):
+                args.method = "gdb"
             if debugger_available("lldb", verbose=verbose):
                 args.method = "lldb"
             elif debugger_available("gdb", verbose=verbose):
```

Comment:
And this will wind up checking for gdb availability twice on Linux if neither debugger is available, so structurally this isn't ideal.

This is actually the perfect case for a Python [`for/else` loop](https://www.geeksforgeeks.org/using-else-conditional-statement-with-for-loop-in-python/). I think we should be able to do something like:
```py
        if args.method == "auto":
            # Prefer gdb on Linux but lldb on macOS
            if platform.system() == "Linux":
                debuggers = ("gdb", "lldb")
            else:
                debuggers = ("lldb", "gdb")

            for debugger in debuggers:
                if debugger_available(debugger, verbose=verbose):
                    args.method = debugger
                    break
            else:
                raise MemrayCommandError(
                    "Cannot find a supported lldb or gdb executable.",
                    exit_code=1,
                )
```

Would you mind testing that and seeing if it does the trick?